### PR TITLE
[CORE] Enhance gluten config parsing

### PR DESCRIPTION
**Pull request by Yohahaha**
## What changes were proposed in this pull request?

Allow parse config like `spark.gluten.sql.columnar.backend.velox.loadQuantum=8m`.

### CODE REVIEW

**Review by FelixYBW — APPROVED** [review]
Sounds good

File: shims/common/src/main/scala/org/apache/spark/sql/internal/GlutenConfigUtil.scala [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import scala.collection.JavaConverters._
+
+object GlutenConfigUtil {
+  def parseConfig(conf: Map[String, String]): Map[String, String] = {
+    val reader = new ConfigReader(conf.filter(_._1.contains(".gluten.")).asJava)
+    val glutenConfigEntries =
+      SQLConf.getConfigEntries().asScala.filter(e => e.key.contains(".gluten."))
+    val glutenConfig = glutenConfigEntries.map(e => (e.key, e.readFrom(reader).toString)).toMap
+    conf.map(e => (e._1, glutenConfig.getOrElse(e._1, e._2)))
+  }
```
Comment (zhztheplayer) [inline shims/common/src/main/scala/org/apache/spark/sql/internal/GlutenConfigUtil.scala:30]: Minor: could it be better to matcn on `spark.gluten.` than on `.gluten.`?

**Review by jackylee-ch — APPROVED** [review]
Gread job. Thanks @Yohahaha

**Review by zhztheplayer — APPROVED** [review]
Thanks!

**Pull request merged**